src/app/services/auth.service.ts

```diff
@@ -0,0 +1,72 @@
+import {Injectable} from '@angular/core';
+import {Router} from '@angular/router';
+import {auth} from 'firebase/app';
+import {Observable} from 'rxjs';
+import {User} from '../models/user.model';
+import * as firebase from 'firebase';
+import {NotificationService} from './notification.service';
+import {of} from 'rxjs/internal/observable/of';
+import {environment} from '../../environments/environment';
+import {AngularFireAuth} from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  user$: Observable<User | null | undefined> = of(undefined);
+
+  constructor(
+    private router: Router,
+    private notificationService: NotificationService,
+    public angularFireAuth: AngularFireAuth
+  ) {
+    if (environment.requireAuth) {
+        angularFireAuth.onAuthStateChanged(firebaseUser => {
+          if (firebaseUser != null) {
+            const user: User = {uid: firebaseUser.uid, displayName: firebaseUser.displayName};
+            this.user$ = of(user);
+          } else {
+            this.user$ = of(null);
+          }
+        });
+      }
+  }
+
+  googleSignin(): void {
+    const provider = new auth.GoogleAuthProvider();
+    this.angularFireAuth.signInWithPopup(provider).then(result => {
```

Comment:
Is there a type annotation you can add for `result` here?

Reply:
Not sure I got an answer to this one?

Reply:
Ah my bad, yes, I added it.